File: Problems/0027.Remove-Element/RemoveElement_test.py
from RemoveElement import Solution
import pytest
from copy import deepcopy

SOL = Solution()

TEST_SUITS =  [
        ([3,2,2,3], 3, 2,  [2, 2]),
        ([0,1,2,2,3,0,4,2], 2, 5, [0, 1, 3, 0, 4]), 
]

@pytest.mark.parametrize(
    "nums, val, ans, val_nums", 
    deepcopy(TEST_SUITS)
)
def test(nums, val, ans, val_nums):
    assert SOL.removeElement(nums, val) == ans
    for i, n in enumerate(val_nums):
        assert nums[i] == n

@pytest.mark.parametrize(
    "nums, val, ans, val_nums", 
    deepcopy(TEST_SUITS)
)
def test_v2(nums, val, ans, val_nums):
    assert SOL.removeElement_v2(nums, val) == ans
    for i, n in enumerate(val_nums):
        assert nums[i] == n

File: Problems/0027.Remove-Element/RemoveElement_test.go
package leetcode

import (
	"testing"
)

func Test_removeElement(t *testing.T) {
	testSuits := []struct {
		name    string
		nums    []int
		val     int
		ans     int
		valNums []int
	}{
		{"Example 1", []int{3, 2, 2, 3}, 3, 2, []int{2, 2}},
		{"Example 2", []int{0, 1, 2, 2, 3, 0, 4, 2}, 2, 5, []int{0, 1, 3, 0, 4}},
	}
	for _, test := range testSuits {
		got := removeElement(test.nums, test.val)

		if got != test.ans {
			t.Errorf("[%v] expect '%v', got '%v' ", test.name, test.ans, got)
		}
		for idx, val := range test.valNums {
			if test.nums[idx] != val {
				t.Errorf("[%v][Validate Array] expect '%v', got '%v' ", test.name, test.valNums, test.nums)
				break
			}
		}
	}
}
